### ISSUE TITLE
Numpy functions, unyt_array, and unyt_quantity

### ISSUE THREAD

**Issue by wattse** [yt-project/unyt#224]
* unyt version: 2.8.0
* Numpy version: 1.21.4
* Python version: 3.9.7
* Operating System: Win10

### Description

I was trying to use np.meshgrid with two unyt_arrays and with one unyt_quantity as inputs. As part of the meshgrid function, there is a comprehension that calls x.copy(), and this causes an error with the unyt_quantity.

Changing the unyt_quantity to a unyt_array eliminates the error.

However, I was expecting the library to handle the differences between array and single value inputs seamlessly, and I expected that I would not have to type check all inputs to the numpy function to ensure that scalars are represented by a 1x1 unyt_array.

### What I Did

```
np.meshgrid(np.array([1., 2., 3.]) * m, np.array([1., 2., 3.]) * m, np.array([1.]) * m)
[unyt_array([[[1.],
 ....]]], 'm'), unyt_array([[[1.],
 ....]]], 'm'), unyt_array([[[1.],
 ....]]], 'm')]

# but
np.meshgrid(np.array([1., 2., 3.]) * m, np.array([1., 2., 3.]) * m, 1. * m)
Traceback (most recent call last):
  File "[\lib\site-packages\numpy\lib\function_base.py]()", line 4373, in meshgrid
    output = [x.copy() for x in output]
  File "[\lib\site-packages\numpy\lib\function_base.py]()", line 4373, in <listcomp>
    output = [x.copy() for x in output]
  File "[\lib\site-packages\unyt\array.py]()", line 1893, in copy
    return type(self)(np.copy(np.asarray(self)), self.units, name=name)
  File "[\lib\site-packages\unyt\array.py]()", line 2065, in __new__
    raise RuntimeError("unyt_quantity instances must be scalars")
RuntimeError: unyt_quantity instances must be scalars

```

**Comment by wattse** [comment]
A further comment on the issue of unyt_quantity and unyt_array:
When performing a direct assignment from a 1x1 unyt_array to a variable, the type of the unyt_array is preserved:

~~~
temp = np.array([1.]) * m
# temp is unyt_array([1.], 'm')
~~~

It looks like any arithmetic operation on the array with another quantity  will convert the 1x1 array to a quantity:

~~~
temp = 2*(np.array([1.]) * m)
# temp is unyt_quantity([2.], 'm')
~~~

**Comment by neutrinoceros** [comment]
unyt_quantity objects are really just 0-D arrays, meaning that `np.array([1.]) * m` is fundamentally different than `np.array(1.) * m` or `1. * m` (these last two are equivalent).
That said, `np.meshgrid` works as you'd expect with 0D arrays, so I agree that the example you gave _should_ work with unyt_quantity as it does with a single-value 1D unyt_array.


> It looks like any arithmetic operation on the array with another quantity will convert the 1x1 array to a quantity:

This looks like a separate bug to me. The shape on the array should certainly not be changed as a side effect of this arithmetic operation.

**Comment by wattse** [comment]
I agree that there are two separate bugs: conversion to unyt_quantity, and handling of unyt_quantity in some numpy functions (at least in meshgrid).

~~~
import numpy as np
from unyt import m

a = np.array(0.1)

np.meshgrid(np.array([1, 2, 3]), np.array([1, 2, 3]), a)
# this will work.

np.meshgrid(np.array([1, 2, 3]), np.array([1, 2, 3]), a * m)
# this causes the same error trace as my original post.
~~~

**Comment by neutrinoceros** [comment]
@wattse I found a fix for both problems in #225 . With that branch

```python
import numpy as np
from unyt import m

a = np.array(1)
print(np.meshgrid(np.array([1, 2]), a * m))
```
prints
```
[array([[1, 2]]), unyt_array([[1, 1]], 'm')]
```

is this what you'd expect ?
